network/p2p/http.go

```diff
@@ -75,13 +75,38 @@ func (s *HTTPServer) RegisterHTTPHandlerFunc(path string, handler func(http.Resp
 	})
 }
 
+type httpClientConfig struct {
+	host host.Host
+}
+
+type httpClientOption func(*httpClientConfig)
+
+// WithHost sets the libp2p host for the http client.
+func WithHost(h host.Host) httpClientOption {
+	return func(o *httpClientConfig) {
+		o.host = h
+	}
+}
+
 // MakeHTTPClient creates a http.Client that uses libp2p transport for a given protocol and peer address.
-func MakeHTTPClient(addrInfo *peer.AddrInfo) (*http.Client, error) {
-	clientStreamHost, err := libp2p.New(libp2p.NoListenAddrs)
-	if err != nil {
-		return nil, err
+// If service is nil, a new libp2p host is created.
+func MakeHTTPClient(addrInfo *peer.AddrInfo, opts ...httpClientOption) (*http.Client, error) {
+	var config httpClientConfig
+	for _, opt := range opts {
+		opt(&config)
+	}
+
+	var clientStreamHost host.Host
+	if config.host != nil {
```

Comment:
Should there be a test validating this?

Reply:
It is implicitly tested via MakeHTTPClientWithRateLimit. The other side of the condition (nil) is also covered in unit tests calling MakeHTTPClient directly.